### PR TITLE
Add Functional Options Pattern and Packages content to the Golang Style Guide

### PR DESCRIPTION
**Pull request by robmorgan**
This PR adds content about the functional options pattern and packages to the Golang style guide.

### CODE REVIEW

File: _docs-sources/guides/style/golang-style-guide.md [hunk 1]
```diff
@@ -181,6 +181,20 @@ Since many languages use `ALL_CAPS` for constants, it is worth calling out expli
...
 `REGION` or `TEST_REGION`.
 
+
+### Using packages
+
+A common Go pattern is creating new packages under the `pkg` or `internal` directories. If you are unsure about which
+one to use, follow this simple litmus test:
+
+- The `internal` directory contains code that we don't want to be consumed by other clients.
+- Telemetry is a great example, as it is typically app-specific and may have a private DSN.
```
Comment (infraredgirl) [inline _docs-sources/guides/style/golang-style-guide.md:191]: Nit: indent this bullet point one level under the one on line 190.

File: _docs-sources/guides/style/golang-style-guide.md [hunk 2]
```diff
@@ -181,6 +181,20 @@ Since many languages use `ALL_CAPS` for constants, it is worth calling out expli
...
+- It would be unfortunate for another CLI app to import this package and inadvertently emit events.
```
Comment (infraredgirl) [inline _docs-sources/guides/style/golang-style-guide.md:192]: Nit: indent this bullet point one level under the one on line 190.

File: _docs-sources/guides/style/golang-style-guide.md [hunk 3]
```diff
@@ -181,6 +181,20 @@ Since many languages use `ALL_CAPS` for constants, it is worth calling out expli
...
+- The `pkg` directory contains all packages that we want to make exportable to other clients.
+- Most of our code should generally live under `pkg`, except for very particular private methods.
```
Comment (infraredgirl) [inline _docs-sources/guides/style/golang-style-guide.md:194]: Nit: indent this bullet point one level under the one on line 193.

**Review by infraredgirl — APPROVED** [review]
LGTM. I left a few nits; they're not blockers to merging this PR.

File: _docs-sources/guides/style/golang-style-guide.md (hunk 1 above)
Comment (robmorgan) [reply _docs-sources/guides/style/golang-style-guide.md]: fixed.

File: _docs-sources/guides/style/golang-style-guide.md (hunk 3 above)
Comment (robmorgan) [reply _docs-sources/guides/style/golang-style-guide.md]: fixed.

File: _docs-sources/guides/style/golang-style-guide.md (hunk 2 above)
Comment (robmorgan) [reply _docs-sources/guides/style/golang-style-guide.md]: fixed.

**Review by infraredgirl — APPROVED** [review]

**Pull request merged**